server.js

```diff
@@ -88,6 +96,40 @@ app.get("/", function (req, res) {
     }
 });
 
+app.use('/uploads', express.static('./uploads', {
+  setHeaders: (res, requestPath) => {
+      let noExtension = !Boolean(path.extname(requestPath));
+      if (noExtension) res.setHeader('Content-Type', 'text/html');
+    }}));
+
+app.post("/", async (req, res) => {
+    try {
+        if (!req.files) {
+            res.send({
+                status: 500,
+                message: 'No file uploaded'
+            });
+        } else {
+            const file = req.files.file,
+                  baseUrl = req.protocol + "://" + req.headers.host + '/',
+                  params = Object.keys(req.body).map(key => key + '=' + req.body[key]).join('&');
```

Comment:
```suggestion
                  params = req.body ? Object.keys(req.body).map(key => key + '=' + req.body[key]).join('&') : "";
```

Otherwise, if there’s no request body, `req.body` will be undefined, and the `Object.keys(req.body)` call is gonna fail.

And for a simple case like `curl -F "file=@index.html" http://localhost/?type=respec` — where the only `-F` value specified is just for the file to upload, but no other params are given — then the request won’t have any request body, so that would cause the `Object.keys(req.body)` call to throw.

Reply:
:+1: